src/FormField/AutoComplete.php

```diff
@@ -128,10 +128,15 @@ public function getData()
             }
         }
 
-        $data = $this->model->export([$this->model->id_field, $this->model->title_field]);
-
+        //fix #327: https://github.com/atk4/ui/issues/327
+        $data = [];
+        $res = $this->model->export([$this->model->id_field, $this->model->title_field]);
+        foreach ($res as $item) {
+            $data[] = ['id' => $item[$this->model->id_field], 'name' => $item[$this->model->title_field]];
+        }
+        
         if ($this->empty) {
-            array_unshift($data, [$this->model->id_field => 0, $this->model->title_field => $this->empty]);
+            array_unshift($data, ['id' => 0, 'name' => $this->empty]);
```

Comment:
id = 0 could have another meaning. Maybe you need `id = ''` here?

Reply:
i think @ibelar made it so id=0 works just fine.

Reply:
You know, theoretically there can be models with id=0 as real id value. Theoretically :)
OK, need to test this.

Reply:
just took over id => 0 from the original implementation. But you are right, a real ID could be 0.

Reply:
yes, leave it, it is a separate issue, you are not making things worse.

Reply:
ok i agree